### ISSUE TITLE
Can you guys reproduce the performance??

### ISSUE THREAD

**Issue by lhtht77** [MultimediaTechLab/YOLO#203]
I trained yolov9 with coco dataset.
The detector's performance stuck on about 36AP.

The official code goes up to about 51 AP.

The difference is so big and the root cause is not clear.

Anybody overcome this issue before?

**Comment by sam31046** [comment]
You could try augmentation

**Comment by BitCode-in** [comment]
Hi everyone,

I had the exact same issue with low metrics on a custom dataset (small objects on complex backgrounds) and wanted to share what actually fixed it for me.

@lhtht77 If your repo doesn't support color augs from the yaml, you have to write a custom transform using torchvision. Here is a quick example for brightness that you can add to your data augmentation script:

```python
import torchvision.transforms.functional as TF
import numpy as np
import torch

class RandomBrightness:
    def __init__(self, prob=0.5, factor_range=[0.8, 1.2]):
        self.prob = prob
        self.factor_range = factor_range

    def __call__(self, image, boxes):
        if torch.rand(1) < self.prob:
            factor = np.random.uniform(self.factor_range[0], self.factor_range[1])
            image = TF.adjust_brightness(image, factor)
        return image, boxes
```
Just make sure your yaml parser unpacks the dict parameters correctly so `prob` doesn't accidentally get passed as a config object.

To save you all some time, here is my complete train.yaml. Switching from SGD to AdamW (with lr 0.001) fixed the convergence speed for me. 

But I also want to highly recommend looking at the scheduler block. Switching to CosineAnnealingLR instead of Linear was another massive leap for my metrics.

```yaml
task: train

defaults:
  - validation: ../validation

epoch: 500

data:
  batch_size: 16
  equivalent_batch_size: 64
  image_size: ${image_size} 
  cpu_num: ${cpu_num}
  shuffle: True
  pin_memory: True
  
  data_augment:
    Mosaic: 1.0
    MixUp: 0.1
    HorizontalFlip: 0.5
    RandomCrop: 0.0
    VerticalFlip: 0.0
    
    RandomBrightness:
      prob: 0.5
      factor_range: [0.6, 1.4]
    RandomContrast:
      prob: 0.5
      factor_range: [0.6, 1.4]
    RandomSaturation:
      prob: 0.5
      factor_range: [0.6, 1.4]
    
    RemoveOutliers: 1e-8

optimizer:
  type: AdamW
  args:
    lr: 0.001
    weight_decay: 0.05
    betas: [0.9, 0.999]

loss:
  objective:
    BCELoss: 0.5
    BoxLoss: 7.5
    DFLoss: 1.5
  aux: 0.25
  matcher:
    iou: CIoU
    topk: 10
    factor:
      iou: 6.0
      cls: 0.5

scheduler:
  type: CosineAnnealingLR
  warmup:
    epochs: 3.0
  args:
    T_max: ${task.epoch}
    eta_min: 0.0001

ema:
  enable: true
  decay: 0.9999
```

Hope this helps you guys get those metrics up.
